### PR TITLE
Initial dev

### PR DESCRIPTION
**Pull request by jbrestel**
moving relevant bits of code from InsertMassSpecFeaturesAndSummaries plugin here

### CODE REVIEW

File: bin/massSpecPeptides.pl [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+  }
+}
+
+
+sub searchProteinSeq {
+  my ($seq, $recordSet, $peptideSequencesToRecords) = @_;
+
+  my %proteinAtts;
+  my $desc = $seq->desc();
+  foreach my $pair (split(/\s/, $desc)) {
```
Comment (bobular) [inline bin/massSpecPeptides.pl:223]: Not a show-stopper but the A. gambiae PEST annotatedProteins.fasta has pipe separators too:
```
>AGAP004677.P163 | transcript=AGAP004677.R163 | gene=AGAP004677 | organism=Anopheles_gambiae_PEST | gene_product=methylenetetrahydrofolate dehydrogenase(NAD+) / 5,10-methenyltetrahydrofolate | transcript_product=methylenetetrahydrofolate dehydrogenase(NAD+) / 5,10-methenyltetrahydrofolate | location=AgamP4_2L:157496-159356(-) | protein_length=303 | sequence_SO=chromosome | SO=protein_coding_gene | is_pseudo=false
```

File: bin/massSpecPeptides.pl [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+  return \%res;
+}
+
+sub getPeptidesFromRecords {
+  my ($records) = @_;
+
+  my %peptides;
+
+  foreach my $record (@$records) {
+    foreach my $peptide ({$record->{peptides}}) {
```
Comment (bobular) [inline bin/massSpecPeptides.pl]:
```suggestion
    foreach my $peptide (@{$record->{peptides}}) {
```

Doesn't work without this!

File: bin/massSpecPeptides.pl [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+      if(!$peptideSequence || $peptideSequence =~ /\d/) {
+        next;# TODO:  why not die here instead??
+      }
+      $record->addPeptide($peptide);
+      # NOTE:  I'm allowing for multiple records having the same id here
+      # This shouldnt' happen but some of the input files used alias mappings so it might?
+      push @{$peptideSequencesToRecords{$peptideSequence}}, $record;
+    } elsif($state eq 'residue') {
+      $residue = MSResidue->new($_);
+      $peptide->addResidue($residue);
```
Comment (bobular) [inline bin/massSpecPeptides.pl:420]: would it be better to call these 'modifiedResidues'? I was confused for a while when I saw in PlasmoDB files that these are not regular amino acid residues, but are acetylations etc.

**Review by bobular — COMMENTED** [review]
I've looked through most of the logic and run the code, added a few debugging statements etc and it all looks good.

I'm not 100% convinced by the overall approach however, which is to take each annotated protein sequence (e.g. from https://vectorbase.org/common/downloads/release-68/AgambiaePEST/fasta/data/VectorBase-68_AgambiaePEST_AnnotatedProteins.fasta) and ask what's the best "peptide set" record from the MS data.

Don't we want to flip it and take each peptide individually (e.g. break up the record) and simply map it to the annotated protein set in a similar way to RNA-Seq mapping? i.e. throw it away if there are multiple hits to different proteins?

File: bin/massSpecPeptides.pl (hunk 1 above)
Comment (jbrestel) [reply bin/massSpecPeptides.pl:223]: this is the internal AnnotatedProteins.fsa file the workflow makes

**Pull request merged**